src/shell/pipe_exec/command_not_found.rs

```diff
@@ -0,0 +1,35 @@
+use super::super::{Capture, Function, Shell};
+use parser::shell_expand::expand_string;
+use std::io::Read;
+use std::process;
+use sys;
+
+pub(crate) fn command_not_found(shell: &mut Shell, command: &str) -> Option<String> {
+    let function = match shell.functions.get("COMMAND_NOT_FOUND") {
+        Some(func) => func as *const Function,
+        None => return None // TODO: Use ? on Option whenever we drop support for older rust versions
+    };
+
+    let mut output = None;
+
+    match shell.fork(Capture::Stdout, |child| unsafe {
+        let _ = function.read().execute(child, &["ion", command]);
```

Comment:
What happens when a function takes the wrong number of arguments? Currently it looks like it just does nothing, but maybe that's not optimal. Should I look into this?

Reply:
It should print an error that an invalid number of arguments was passed / certain arguments had an invalid type.